refactor: Extract linkifier non-settings logic from `markdown.js`.

akshatdalton opened the pull request:
- 1st commit extracts the linkifier non-settings logic from `markdown.js`.

- And the 2nd commit merges the `markdown_katex.js` test file with `markdown.js` test file.
We split them in the commit: https://github.com/zulip/zulip/commit/c50dbf82974cafc4eeb00c5d1fe86ab18e000374 because at that time, we thought markdown_katex was gonna require rewireproxy magic in the later commits. But we later removed the rewiremock dependency in the commit: https://github.com/zulip/zulip/commit/30c7108955d86d9b860ef6e746ef1e707dbfe62f.

   So now, we can safely merge the `markdown_katex.js` test file with the `markdown.js` test file.

timabbott commented on static/js/realm_linkifiers.js:
```diff
@@ -0,0 +1,109 @@
+import marked from "../third/marked/lib/marked";
```
I think we want to call this `linkifiers.js`; when we add stream-level linkifiers in the future, we'll likely want them to be managed by the same file.

akshatdalton commented:
Renamed the new file `realm_linkifiers.js` -> `linkifiers.js` and updated the commit message explaining why we name it `linkifiers.js` and not `realm_linkifiers.js`.

@timabbott, please review.

timabbott commented:
Merged, thanks @akshatdalton!
